File: src/main/js/formulas/info/ISNUMBER.ts
import { SheetColumnRowKey } from "../../models/common/SheetColumnRowKey";
import { Complex } from "../../models/common/Types";
import { Converters } from "../../utils/Converters";
import { AbstractFormula } from "../AbstractFormula";
import { FormulaName } from "../FormulaName";

export class ISNUMBER extends AbstractFormula {
  static SELF: ISNUMBER = new ISNUMBER();
  NAME = FormulaName.ISNUMBER;

  internal(origin: SheetColumnRowKey, ...values: Array<Complex>) {
    AbstractFormula.checkLength(values.length, 1, this.NAME);
    const first = Converters.first(this.collateralLookup(origin, values[0]));
    return typeof first === "number";
  }
}

File: src/main/java/io/protobase/f7/formulas/info/ISNUMBER.java
package io.protobase.f7.formulas.info;

import io.protobase.f7.formulas.AbstractFormula;
import io.protobase.f7.formulas.FormulaName;
import io.protobase.f7.models.GridColumnRowKey;
import io.protobase.f7.utils.Converters;

import java.util.function.BiFunction;
import java.util.function.Function;

public class ISNUMBER extends AbstractFormula {
  public static FormulaName NAME = FormulaName.ISNUMBER;
  public static ISNUMBER SELF = new ISNUMBER();

  public ISNUMBER() {
    super();
  }

  public ISNUMBER(Function<Object, Object> lookup, BiFunction<GridColumnRowKey, Object, Object> collateralLookup) {
    super(lookup, collateralLookup);
  }

  @Override
  public Object internal(GridColumnRowKey origin, Object... values) {
    checkLength(values.length, 1, NAME);
    return Converters.first(collateralLookup.apply(origin, values[0])) instanceof Double;
  }
}
